src/Monolog/Handler/FormattableHandlerTrait.php

```diff
@@ -22,9 +22,9 @@
 trait FormattableHandlerTrait
 {
     /**
-     * @var FormatterInterface
+     * @var ?FormatterInterface
      */
-    protected $formatter;
+    protected $formatter = null;
```

Comment:
this change is useless. It is already the case in PHP

Reply:
With an untyped property, yes, the `null` default is implicit.

If this were formally a PHP 7.4 typed property `protected ?FormatterInterface $formatter`, then `FormattableHandlerTrait::getHandler` would throw an `E_ERROR` due to the access of an uninitialized property. Example [here](https://3v4l.org/O0TgZ).

So my reasoning is as follows:
* Under PHP < 7.4 and/or not using typed properties, this change is _redundant_ but not _wrong_.
* Under PHP >= 7.4 with typed properties, it is no longer _redundant_ and is in fact _necessary_.
* On first inspection, there doesn't seem to be any difference in the compiled opcodes between the two versions.
* Hence, adding this initialization implicitly gives you forward-compatibility if you choose to implement typed properties in some future iteration of the library, and at least appears to not have any runtime impact.

This is bikeshedding to the extreme, I know. I am not attached to the explicit instantiation here in any case: what I'm really after is the type annotation that `$formatter` is nullable so Psalm level 2 will quit barking about my custom log handler not being initialized :smiley: .

Reply:
I'm also seeing this issue in v7.4.11.  If the fix operates as claimed it'd be beneficial to have this.

Reply:
but this trait does not use typed properties.
And I did not say that the change was wrong, but that it was useless (and inconsistent with the coding standards used throughout the package)

Reply:
That's fine then, I can remove it

e87d182